apps/stops/lib/route_stops.ex

```diff
@@ -46,7 +46,7 @@ defmodule Stops.RouteStops do
   end
 
   @spec from_list([RouteStop.t()]) :: t()
-  def from_list(stops) do
+  def from_list(stops) when stops != [] do
```

Comment:
I think this will cause the same FunctionClauseError when stops _do_ equal [], just further up the call stack.  If stops is [], then there will be no function match for `from_list/1`.